pkg/process/heartbeat/monitor.go

```diff
@@ -0,0 +1,144 @@
+package heartbeat
+
+import (
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/process/net"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-go/statsd"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// ModuleMonitor is responsible for emitting heartbeat metrics for each
+// system-probe module. It does so by hitting the stats endpoint from
+// system-probe and emitting one metric per enabled module using Datadog Public
+// API.
+type ModuleMonitor struct {
+	statsFn statsFn
+	flusher flusher
+	exit    chan struct{}
+}
+
+// Options encapsulates all configuration params used by ModuleMonitor
+type Options struct {
+	// KeysPerDomain (required) contains API key entries per Datadog domain
+	KeysPerDomain map[string][]string
+
+	// HostName (required)
+	HostName string
+
+	// SysprobeSocketPath (optional) sets the location of the Unix socket used
+	// to reach system-probe
+	SysprobeSocketPath string
+
+	// StatsdClient (optional) points to a statsd client to be used as a
+	// fallback mechanism
+	StatsdClient statsd.ClientInterface
+
+	// TagVersion (optional) contains the agent version to be sent along with the
+	// hearbeat metric
+	TagVersion string
+
+	// TagRevision (optional) contains the agent revision to be sent along with the
+	// hearbeat metric
+	TagRevision string
+}
+
+// statsFn represents the function signature used to fetch stats from system-probe
+type statsFn func() (map[string]interface{}, error)
+
+// NewModuleMonitor returns a new ModuleMonitor
+func NewModuleMonitor(opts Options) (*ModuleMonitor, error) {
+	if opts.SysprobeSocketPath != "" {
+		net.SetSystemProbePath(opts.SysprobeSocketPath)
+	}
+
+	flusher, err := newStatsdFlusher(opts)
+	if err != nil {
+		log.Warnf("could not create statsd flusher: %s", err)
+	}
+
+	apiFlusher, err := newAPIFlusher(opts)
+	if err != nil {
+		log.Warnf("could not create api flusher: %s", err)
+	} else {
```

Comment:
Does this 'healthcheck' the endpoint ?

Reply:
I'll resolve this since the healthchecking logic is now being done by the `apiWatcher`